openpype/hosts/aftereffects/api/extension/jsx/hostscript.jsx

```diff
@@ -364,7 +370,22 @@ function importFileWithDialog(path, item_name){
 
 
 function getCompFilepath(compItem){
-    return String(compItem.layers[1].source.file)
+    for (var index = 1; index <= compItem.numLayers; index++) {
+        // search if source.file is available aka the layer is not a comp
+        //if one is present , it returns the path
+        if (compItem.layers[index].source.file){
+            return String(compItem.layers[index].source.file)
+        }
+    }
+    //else if none is present, must search in the imported folder of AE for layer.
+    folder = getImportedCompFolder(compItem)
+    for (var index = 1; index <= folder.items.length; index++) {
```

Comment:
There is no fallback return value if we never enter in the if condition

Reply:
It was previously the case, I don't really know what to return if we never enter in the if condition

Reply:
I don't know, either the previous code value: `return String(compItem.layers[1].source.file)` or an empty string `return String("")` or an error, but at least a comment to state that this should not happend

Reply:
At the line 343, a condition has been added to detected if nothing is returned to trigger a Error